### PR TITLE
Refactor Repositories and Implement Deck Initialization and Draw Card Features

### PR DESCRIPTION
**Pull request by linycken013127**
## Why need this change? / Root cause:
- The current responsibilities of the PlayerRepository and CardRepository functions exceed those of a typical Repository.
- The current Schema leads to redundant storage of the same card information in the game card.
- The deck initialization and card distribution functions are implemented when the game starts.

## Changes made:
- Unified file naming conventions.
- Standardized receiver names.
- Refactored PlayerRepository and CardRepository.
- Normalized the Card table by splitting out a Deck table.
- Implemented deck initialization and draw card functionalities.

## Test Scope / Change impact:
- Acceptance tests added to verify that, after drawing the cards, all players have received their game cards.

## Issue:
- https://github.com/Game-as-a-Service/The-Message/issues/38

### CODE REVIEW

File: Backend/service/repository/mysql/card_repository.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	result := c.db.Table("cards").Create(card)
+
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return card, nil
+}
+
+func (c *CardRepository) Get(ctx context.Context) ([]*repository.Card, error) {
```
Comment (ping-yee) [inline Backend/service/repository/mysql/card_repository.go:42]: Or should the naming of this method be `GetCards`?

File: Backend/service/repository/mysql/deck_repository.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	result := d.db.Table("decks").First(deck, "id = ?", id)
+
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return deck, nil
+}
+
+func (d *DeckRepository) Get(ctx context.Context) ([]*repository.Deck, error) {
```
Comment (ping-yee) [inline Backend/service/repository/mysql/deck_repository.go:31]: As above.

**Review by ping-yee — CHANGES_REQUESTED** [review]

**Comment by linycken013127** [comment]
@ping-yee Review, please.

**Review by ping-yee — APPROVED** [review]
LGTM

**Pull request merged**